ci_framework/roles/hive/tasks/openstack_cluster_claim.yml

```diff
@@ -0,0 +1,76 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+# Claims a cluster from a the provided pool
+-
+  name: Claim an OpenShift Cluster
+  hosts: localhost
+  vars:
+    claim_name: "{{ cifmw_hive_openstack_claim_name }}"
+    cluster_pool: "{{ cifmw_hive_openstack_pool_name }}"
+    rtn_kube: "{{ cifmw_hive_kubeconfig }}"
+    life_time: "{{ cifmw_hive_lifetime | default(false) }}"
+    namespace: "{{ cifmw_hive_namespace }}"
+
+  tasks:
+    - name: Setting oc_cmd variable
+        ansible.builtin.set_fact:
+          oc_cmd: "oc --kubeconfig {{ cifmw_hive_kubeconfig }}"
+        when: cifmw_hive_kubeconfig is defined
+
+    - name: Check if ClusterPool exists
+      ansible.builtin.shell:
+        cmd: "{{ oc_cmd }} get ClusterPool {{ cluster_pool }} -n {{ namespace }}"
+
+    - name: Creating the cluster claim
+      ansible.builtin.include_tasks:
+        file: oc-apply.yml
+      vars:
+        desc: "cluster-claim"
+        destFile: "cluster_claim.yml"
+        oc_cli: "{{ oc_cmd }}"
+        templateFile: "templates/claim.yml.j2"
+        workingDir: "{{ oc_cli_space.path }}"
```

Comment:
As we are not using `oc_cli_space.path`, we could switch to using `workdir | cifmw_workdir`or create one during runtime.